executor/aggfuncs/func_avg.go

```diff
@@ -285,11 +383,42 @@ func (e *avgOriginal4Float64) UpdatePartialResult(sctx sessionctx.Context, rowsI
 	return nil
 }
 
-type avgPartial4Float64 struct {
+type avgOriginal4Float64 struct {
+	avgOriginal4Float64HighPrecision
+}
+
+func (e *avgOriginal4Float64) Slide(sctx sessionctx.Context, rows []chunk.Row, lastStart, lastEnd uint64, shiftStart, shiftEnd uint64, pr PartialResult) error {
+	p := (*partialResult4AvgFloat64)(pr)
+	for i := uint64(0); i < shiftEnd; i++ {
+		input, isNull, err := e.args[0].EvalReal(sctx, rows[lastEnd+i])
+		if err != nil {
+			return err
+		}
+		if isNull {
+			continue
+		}
+		p.sum += input
+		p.count++
+	}
+	for i := uint64(0); i < shiftStart; i++ {
+		input, isNull, err := e.args[0].EvalReal(sctx, rows[lastStart+i])
+		if err != nil {
+			return err
+		}
+		if isNull {
+			continue
+		}
+		p.sum -= input
+		p.count--
+	}
+	return nil
+}
+
+type avgPartial4Float64HighPrecision struct {
```

Comment:
Window function will only be used as `aggregation.CompleteMode`, so that this will never be hit.

Reply:
I will remove them.